src/main/webapp/index-script.js

```diff
@@ -303,6 +302,24 @@ function handleSearchResults(results, service) {
   });
 }
 
+/** Deletes all markers in the array by removing references to them. */
+function deleteMarkers() {
+  clearMarkers();
+  markers = [];
+}
+
+/** Removes the markers from the map, but keeps them in the array. */
+function clearMarkers() {
+  setMapOnAllNull();
+}
+
+/** Sets the map on all markers in the array. */
+function setMapOnAllNull() {
+  for (let i = 0; i < markers.length; i++) {
+    markers[i].setMap(null);
+  }
+}
+
```

Comment:
Can we make markers be an input to these functions instead of just operating on an existing variable. It's good practice to limit the scope.